### ISSUE TITLE
`mailto:` link with multiple emails (and potentially a subject) isn't handled correctly

### ISSUE THREAD

**Issue by DanielYang59** [tcort/markdown-link-check#378]
It looks `mailto:` link with multiple emails (and potentially a subject) [might not have been handled correctly](https://github.com/materialsproject/pymatgen/actions/runs/12078446868/job/33682870202?pr=4201), 

The links has the following format (I removed the real email addresses for privacy concerns, but if there emails addresses are necessary to recreate this you could [check here](https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/README.md?plain=1#L92), it's public):
```
mailto:user_a@address,user_b@gmail.com,user_c@another.address?subject=someSubjectOfTheEmail
```

And I got the following error:
> [✖] mailto:user_a@address,user_b@gmail.com,user_c@another.address?subject=someSubjectOfTheEmail → Status: 400

I'm not sure if this is an [`isemail`](https://github.com/skeggse/isemail) issue though, but I noticed it's no longer maintained:
>  This repository has been archived by the owner on Oct 2, 2024. It is now read-only.

**Comment by MikeMcC399** [comment]
@DanielYang59 

> I'm not sure if this is an [isemail](https://github.com/skeggse/isemail) issue though, but I noticed it's no longer maintained:

- See https://github.com/tcort/markdown-link-check/issues/371

**Comment by DanielYang59** [comment]
Hi Mike,

Thanks for letting me know. I don't know much about the internal working of `markdown-link-check ` so not sure if I should propagate the issue to actual worker (`isemail` or `node-email-verifier` or whatever)

**Comment by MikeMcC399** [comment]
@DanielYang59 

> Thanks for letting me know. I don't know much about the internal working of `markdown-link-check ` so not sure if I should propagate the issue to actual worker (`isemail` or `node-email-verifier` or whatever)

I'm not familiar with this in-depth, but It's probably an enhancement request for this repo. You can check yourself about the functionality of [node-email-verifier](https://www.npmjs.com/package/node-email-verifier), which looks like it only deals with a single e-mail address and not a complete `mailto:` link with multiple addresses and a subject field.

**Comment by smainil** [comment]
the code of the email check is here https://github.com/tcort/link-check/blob/3263c3b65917375c209928ca2b1ad4102824da36/lib/proto/mailto.js#L8. It only checks the text between the `mailto:` and the `?` character as a single email. It should try to split the string using the comma separator and loop on each single email found

**Comment by MikeMcC399** [comment]
Based on https://github.com/tcort/markdown-link-check/issues/378#issuecomment-2522348292 this would still be an open issue as the referenced code https://github.com/tcort/link-check/blob/master/lib/proto/mailto.js was last changed in Oct 2024 when isemail was replaced.

**Comment by MikeMcC399** [comment]
This issue is still open and reproducible.

`markdown-link-check` does not respect the syntax of [RFC 6068 - The 'mailto' URI Scheme](https://datatracker.ietf.org/doc/html/rfc6068) for multiple e-mail addresses, and logs a `400` error.

```shell
cat > multiple-mail.md <<EOT
[send to one recipient](mailto:no-user1@gmail.com)
[send to two recipients](mailto:no-user1@gmail.com,no-user2@gmail.com)
EOT
npm install markdown-link-check@latest -g
markdown-link-check multiple-mail.md
```

```text
FILE: multiple-mail.md
  [✓] mailto:no-user1@gmail.com
  [✖] mailto:no-user1@gmail.com,no-user2@gmail.com

  2 links checked.

  ERROR: 1 dead links found!
  [✖] mailto:no-user1@gmail.com,no-user2@gmail.com → Status: 400
```

**Comment by MikeMcC399** [comment]
The last release was [v3.13.7](https://github.com/tcort/markdown-link-check/releases/tag/v3.13.7) on March 11, 2025 which is 6 months ago.

I suggest to increase the stale-bot time-frame accordingly. There is no point in flagging an issue as stale after 2 months if the typical release cadence is greater than 2 months. Otherwise issues like this may get automatically closed, even though the bug / feature request is still relevant.

**Comment by DanielYang59** [comment]
Yep I agree it's quite annoying if we need to add a random comment every now and often just to prevent the PR being closed by the bot...

**Comment by DanielYang59** [comment]
I would take on this task myself (instead of having to deal with the bot)....

**Comment by MikeMcC399** [comment]
@tcort / @DanielYang59 

- I believe that this issue has been prematurely closed due to PR https://github.com/tcort/link-check/pull/106 including an auto-close for this issue (https://github.com/tcort/markdown-link-check/issues/378)

- Now, although https://github.com/tcort/link-check/pull/106 resulted in the release of [link-check@5.5.0](https://github.com/tcort/link-check/tree/v5.5.0), this does not automatically fix `markdown-link-check`.

To fix `markdown-link-check` reliably, it would need https://github.com/tcort/markdown-link-check/blob/e6703d0783204e80a31ebe2097b96b3165c3907f/package.json#L39 bumping to `"link-check": "^5.5.0"` and a new release of `markdown-link-check` would need to be cut. If that isn't done, only new installations of `markdown-link-check` will get the fix. Existing installations won't get notified or updated about a fix.
